Automattic/woocommerce-payments
# Fix translations not loading for Javascript files

Fixes #1712 and #1715

### Changes proposed in this Pull Request

A language pack zip file contains a `json file` for each `js file path` present in the original POT file, and that json file has the md5 hash of the file path in its name (ex: `woocommerce-payments-es_ES-c817bb653276f1ca458c82d18b93b92a.json`), but the UI uses bundled files (dist/index.js) so, no json file is loaded for them.

This PR ports the solution used in the `woocommerce-admin` plugin, which consists of combining all the json chunk files into a single `woocommerce-payments-es_ES.json` file after getting new translations (using the action `upgrader_process_complete`), and adds a filter (`load_script_translation_file`) to load it for all `WCPAY_*` script handles.

__NOTE__: The table footer (summary) not translated (#1715) is also fixed, to test it add the following `messages` to the `woocommerce-payments-es_ES.json` file (or similar depending on the locale): `deposits`, `total`, `transactions`, `fees` and `net`.
This is necessary because those `strings` are `new` and new translation strings are uploaded to _GlotPress_ during the release process, so they are not included in the language pack... (they will be included after merging to develop and creating a test release)

### Testing instructions

1. Change the site language to one that has a language pack available (ex: Español)
2. Go to Dashboard > Updates, at the bottom of the page click: `Update Translations`
3. Go to Payments (ex: Pagos), and see that all texts are translated to the selected site language

Before:
<img width="1042" alt="Screen Shot 2021-05-04 at 02 25 12" src="https://user-images.githubusercontent.com/407542/116963738-169b4d00-ac80-11eb-8af8-469c3ead86ec.png">

After:
<img width="1043" alt="Screen Shot 2021-05-04 at 02 24 16" src="https://user-images.githubusercontent.com/407542/116963747-1a2ed400-ac80-11eb-88dd-da959ae03672.png">

-------------------
